src/Workspaces/Core/Portable/Shared/Utilities/StringBreaker.cs

```diff
@@ -275,13 +275,14 @@ private static TextSpan ScanWordRun(string identifier, int length, int wordStart
                 {
                     // hit the 'o' in XMLDo.  Return "XML"
                     Debug.Assert(char.IsUpper(identifier[current - 1]));
-                    return TextSpan.FromBounds(wordStart, current - 1);
```

Comment:
Yes, my expectation if this was a measured performance problem was to eliminate the redundant argument checks, which fixes the problem without making the call sites harder to read.